community/server/src/test/java/org/neo4j/server/HttpHeadersIT.java

```diff
@@ -0,0 +1,239 @@
+import com.sun.jersey.api.client.config.ClientConfig;
+import com.sun.jersey.api.client.config.DefaultClientConfig;
+import com.sun.jersey.client.urlconnection.HTTPSProperties;
+import org.junit.After;
+import org.junit.Test;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+
+import org.neo4j.server.configuration.ServerSettings;
+import org.neo4j.server.helpers.CommunityServerBuilder;
+import org.neo4j.test.server.ExclusiveServerTestBase;
+import org.neo4j.test.server.InsecureTrustManager;
+
+import static com.sun.jersey.client.urlconnection.HTTPSProperties.PROPERTY_HTTPS_PROPERTIES;
+import static java.util.Collections.emptyList;
+import static org.eclipse.jetty.http.HttpHeader.SERVER;
+import static org.eclipse.jetty.http.HttpHeader.STRICT_TRANSPORT_SECURITY;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.neo4j.server.helpers.CommunityServerBuilder.serverOnRandomPorts;
+import static org.neo4j.server.security.ssl.HttpsRequestCustomizer.PUBLIC_KEY_PINS_HTTP_HEADER;
+
+public class HttpHeadersIT extends ExclusiveServerTestBase
+{
+    private static final String HSTS_HEADER_VALUE = "max-age=31536000; includeSubDomains; preload";
+    private static final String HPKP_HEADER_VALUE = "max-age=5184000; " +
+                                                    "pin-sha256=\"d6qzRu9zOECb90Uez27xWltNsj0e1Md7GkYYkVoZWmM=\"; " +
+                                                    "pin-sha256=\"E9CZ9INDbd+2eRQozYqqbQ2yXLVKB9+xcprMF+44U1g=\"; " +
+                                                    "includeSubDomains";
+
+    private CommunityNeoServer server;
+
+    @After
+    public void tearDown() throws Exception
+    {
+        if ( server != null )
+        {
+            server.stop();
+        }
+    }
+
+    @Test
+    public void shouldNotSendJettyVersionWithHttpResponseHeaders() throws Exception
+    {
+        startServer();
+        testNoJettyVersionInResponseHeaders( httpUri() );
+    }
+
+    @Test
+    public void shouldNotSendJettyVersionWithHttpsResponseHeaders() throws Exception
+    {
+        startServer();
+        testNoJettyVersionInResponseHeaders( httpsUri() );
+    }
+
+    @Test
+    public void shouldNotSendHstsHeaderWithHttpResponse() throws Exception
+    {
+        startServer( HSTS_HEADER_VALUE, null );
+        assertNull( runRequestAndGetHstsHeaderValue( httpUri() ) );
+    }
+
+    @Test
+    public void shouldSendHstsHeaderWithHttpsResponse() throws Exception
+    {
+        startServer( HSTS_HEADER_VALUE, null );
+        assertEquals( HSTS_HEADER_VALUE, runRequestAndGetHstsHeaderValue( httpsUri() ) );
+    }
+
+    @Test
+    public void shouldNotSendHstsHeaderWithHttpsResponseWhenMaxAgeNotConfigured() throws Exception
```

Comment:
Should the test name be `shouldNotSendHstsHeaderWithHttpsResponseWhenNotConfigured`?